### PR TITLE
Reflect GameEntityContextRequestBus to BehaviorContext

### LINKED ISSUE #8992: Level load readiness signal for testing
**Is your feature request related to a problem? Please describe.**
The lack of a reliable level load / test-readiness signal results in most tests relying on blindly waiting for in hope that the level will idle. This can be seen in https://github.com/o3de/o3de/blob/6ed9dc74766af825a8b8a3972fbb76b6ffcc6d3d/AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py#L96

This is not only inefficient, it cannot assure readiness in the case where level load takes longer than expected.

**Describe the solution you'd like**
Level load (which can result in iteratively streaming in additional content) provides a signal, callback, or promise mechanism to eliminate blind waits. Ideally this readiness signal is available in both the Editor as well as in Launchers. This will require editor (or test framework) logging that clarifies whether the level loaded, however the logging should not be the primary (or only) readiness signal.

**Describe alternatives you've considered**
Tests can be required to solely rely on setting up levels during execution, and not on loading prebuilt levels.

**Additional context**
* Pain from this inefficiency will significantly increase as more tests load levels.
* This mechanism may require architectural input from sig-core and sig-testing, and may be relevant to sig-content for use in scripting and gameplay.

### PR DESCRIPTION
**Pull request by carlitosan**
Closes #8992 

Signed-off-by: carlitosan <82187351+carlitosan@users.noreply.github.com>

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Code/Framework/AzFramework/AzFramework/Entity/GameEntityContextComponent.cpp [hunk 1]
```diff
@@ -56,6 +56,14 @@ namespace AzFramework
                     ->Attribute(AZ::ScriptCanvasAttributes::DeactivatesInputEntity, true)
                 ->Event("GetEntityName", &GameEntityContextRequestBus::Events::GetEntityName)
                 ;
+
+            behaviorContext->EBus<GameEntityContextEventBus>("GameEntityContextEventBus")
+                ->Attribute(AZ::Script::Attributes::Module, "entity")
+                ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
```
Comment (onecent1101) [inline Code/Framework/AzFramework/AzFramework/Entity/GameEntityContextComponent.cpp:62]: minor: probably you want to include Category so these ebus node can be grouped under Entity?

**Review by onecent1101 — APPROVED** [review]

**Review by Kadino — COMMENTED** [review]
Not quite clear how this resolves https://github.com/o3de/o3de/issues/8992 as these signals appear to be signalling the request/start of load, and not the completion of load. Can you clarify how a test would use this as a "loading complete" signal?

**Comment by lumberyard-employee-dm** [comment]
> Not quite clear how this resolves #8992 as these signals appear to be signalling the request/start of load, and not the completion of load. Can you clarify how a test would use this as a "loading complete" signal?

The [OnGameEntitiesStarted](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Entity/GameEntityContextBus.h#L166-L170) event is signaled after the level has loaded

**Comment by chiyenteng** [comment]
Is this ready to be merged?

**Pull request merged**